ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py

```diff
@@ -1,29 +1,129 @@
+        if not pickle_file.exists():
+            return True
+    # Check if any hash file is missing
+    for id, schema_file in DEFAULT_SCHEMAS.items():
+        if not schema_file.with_suffix(".sha1").exists():
+            return True
+
+    # Check if any hash does not match the has of the schema file
+    for id, schema_file in DEFAULT_SCHEMAS.items():
+        existing_hash = schema_file.with_suffix(".sha1").read_text(encoding="UTF-8")
+        new_hash = sha1(schema_file.read_bytes(), usedforsecurity=False).hexdigest()
+        if existing_hash != new_hash:
+            return True
+
+    return False
+
+
+def _create_store_from_yaml() -> dict[str, dict]:
+    """
+    Returns a schema store loaded from yaml/json files with $ref
+    """
+    store = {}
+    for id, schema_file in DEFAULT_SCHEMAS.items():
+        with open(schema_file, "r", encoding="UTF-8") as stream:
+            store[id] = safe_load(stream)
+    return store
+
+
+def _compile_schemas() -> dict:
+    """
+    Load schemas from yaml files,
+    create a temporary "store",
+    resolve all $refs and save the resulting schemas as pickles
+    """
+    resolved_schema_store = {}
+
+    temp_schema_store = _create_store_from_yaml()
+
+    # We rely on eos_cli_config_gen being before eos_designs,
+    # so anything in eos_cli_config_gen can be resolved and $def popped before resolving from eos_designs.
+    for schema_name, pickle_file in DEFAULT_PICKLED_SCHEMAS.items():
+        if schema_name == "avd_meta_schema":
+            # Do not resolve $ref in the meta schema.
+            resolved_schema = temp_schema_store[schema_name]
+        else:
+            # Copying so we can pop below.
+            resolved_schema = _resolve_schema(temp_schema_store[schema_name].copy(), temp_schema_store)
+
+            # Since the schema is now fully resolved we can drop the $defs.
+            resolved_schema.pop("$defs", None)
+
+            # Inplace update the schema store with the resolved variant without $def.
+            # This is needed so eos_designs will not resolve to a schema with another $ref.
+            temp_schema_store[schema_name] = resolved_schema
+
+        resolved_schema_store[schema_name] = resolved_schema
+
+        # Update pickle file with binary version of the completely resolved schema.
+        try:
+            with pickle_file.open("wb") as stream:
+                pickle_dump(resolved_schema, stream, HIGHEST_PROTOCOL)
+
+            # Update the .sha1 file with the new hash of the yaml schema file.
+            schema_file = DEFAULT_SCHEMAS[schema_name]
+            new_hash = sha1(schema_file.read_bytes(), usedforsecurity=False).hexdigest()
+            schema_file.with_suffix(".sha1").write_text(new_hash, encoding="UTF-8")
+        except PermissionError:
+            # Ignoring PermissionError so we can operate in read-only environments
+            # (like "ansible-test units" containers).
+            pass
```

Comment:
We don't do logging in AVD today.

Reply:
this is just failing silently - which can be hard to troubleshoot

Reply:
Understood - do you think it could be puzzling in some troubleshooting situations maybe? Like seeing the pickle file not being updated silently because force_rebuild is set to True but permission is failing and this is just eaten by the function?

Not willing to block the PR for this but wondering if it could lead to head banging troubleshooting sessions :)

Reply:
force rebuild is only used by the pre-commit / schema tooling scripts, so not something part of AVD. We have to move on and there is no alternative for this available today.